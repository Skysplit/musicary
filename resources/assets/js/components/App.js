import React, { Component } from 'react';

export class App extends Component {
  render() {
    return (
      <div>Test site</div>
    );
  }
}

export default App;
