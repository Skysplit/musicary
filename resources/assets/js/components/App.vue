<script>
const store = require('../store')

module.exports = {
    store,
    router,
    data() {
        return {
        }
    },
    created() {
        this.$auth.commit('ready')
    }
}
</script>
