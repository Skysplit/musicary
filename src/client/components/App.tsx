import React, { PureComponent } from 'react';
import ListItemLink from '@app/client/components/ListItemLink';
import { UserInterface } from '@client/store/user';
import { NavigationDrawer, ListItem } from 'react-md';
import { AppContainerProps } from '@app/client/containers/AppContainer';
import { removeUser } from '@app/client/utils/userData';

export interface AppProps extends AppContainerProps {
  user: UserInterface;
}

export default class App extends PureComponent<AppProps> {
  private logout = () => {
    removeUser();
    this.props.logout();
    this.props.router.push('/');
  }

  private guestItems() {
    return [
      <ListItem
        key="login"
        component={ListItemLink}
        route="login"
        primaryText="Login"
      />,

      <ListItem
        key="signup"
        primaryText="Sign up"
      />,
    ];
  }

  private userItems() {
    return [
      <ListItem
      key="playlists"
      primaryText="My playlists"
      component={ListItemLink}
      route="playlists"
      />,
      <ListItem
        key="logout"
        primaryText="Log out"
        onClick={this.logout}
      />,
    ];
  }

  private items() {
    if (this.props.user) {
      return this.userItems();
    }

    return this.guestItems();
  }


  render() {
    const { children, title } = this.props;


    return (
      <NavigationDrawer
        drawerHeader={false}
        toolbarTitle={title || 'Musicary'}
        navItems={[
          <ListItem
            key="home"
            component={ListItemLink}
            route="home"
            primaryText="Home"
          />,
          ...this.items(),
        ]}
      >
        {children}
      </NavigationDrawer>
    );
  }
}
