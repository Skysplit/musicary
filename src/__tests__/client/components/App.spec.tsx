import React from 'react';
import { shallow } from 'enzyme';
import App from '@client/components/App';

describe('<App />', () => {
  test('should render user interface', () => {
    const wrapper = shallow(<App user={{ id: 'test' }} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render guest interface', () => {
    const wrapper = shallow(<App user={null} />);
    expect(wrapper).toMatchSnapshot();
  });
});
