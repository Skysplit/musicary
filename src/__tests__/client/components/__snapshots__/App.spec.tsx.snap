// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render guest interface 1`] = `
<NavigationDrawer
  autoclose={true}
  constantDrawerType={false}
  contentComponent="main"
  contentId="main-content"
  defaultMedia="mobile"
  desktopDrawerType="full-height"
  desktopMinWidth={1025}
  drawerTitle="Musicary"
  drawerTransitionDuration={300}
  extractMini={true}
  includeDrawerHeader={true}
  jumpLabel="Jump to content"
  mobileDrawerType="temporary"
  mobileMinWidth={320}
  navItems={
    Array [
      <ListItem
        activeClassName="md-text--theme-primary"
        animateNestedItems={true}
        component={[Function]}
        expanderIcon={
          <FontIcon
            iconClassName="material-icons"
          >
            keyboard_arrow_down
          </FontIcon>
        }
        itemComponent="li"
        primaryText="Home"
        route="home"
      />,
      <ListItem
        activeClassName="md-text--theme-primary"
        animateNestedItems={true}
        component={[Function]}
        expanderIcon={
          <FontIcon
            iconClassName="material-icons"
          >
            keyboard_arrow_down
          </FontIcon>
        }
        itemComponent="li"
        primaryText="Login"
        route="login"
      />,
      <ListItem
        activeClassName="md-text--theme-primary"
        animateNestedItems={true}
        component={[Function]}
        expanderIcon={
          <FontIcon
            iconClassName="material-icons"
          >
            keyboard_arrow_down
          </FontIcon>
        }
        itemComponent="li"
        primaryText="Sign up"
        route="signup"
      />,
    ]
  }
  persistentIcon={
    <FontIcon
      iconClassName="material-icons"
    >
      arrow_back
    </FontIcon>
  }
  position="left"
  tabletDrawerType="persistent"
  tabletMinWidth={768}
  temporaryIcon={
    <FontIcon
      iconClassName="material-icons"
    >
      menu
    </FontIcon>
  }
  toolbarThemeType="colored"
  toolbarTitle="Musicary"
  transitionEnterTimeout={300}
  transitionName="md-cross-fade"
/>
`;

exports[`<App /> should render user interface 1`] = `
<NavigationDrawer
  autoclose={true}
  constantDrawerType={false}
  contentComponent="main"
  contentId="main-content"
  defaultMedia="mobile"
  desktopDrawerType="full-height"
  desktopMinWidth={1025}
  drawerTitle="Musicary"
  drawerTransitionDuration={300}
  extractMini={true}
  includeDrawerHeader={true}
  jumpLabel="Jump to content"
  mobileDrawerType="temporary"
  mobileMinWidth={320}
  navItems={
    Array [
      <ListItem
        activeClassName="md-text--theme-primary"
        animateNestedItems={true}
        component={[Function]}
        expanderIcon={
          <FontIcon
            iconClassName="material-icons"
          >
            keyboard_arrow_down
          </FontIcon>
        }
        itemComponent="li"
        primaryText="Home"
        route="home"
      />,
      <ListItem
        activeClassName="md-text--theme-primary"
        animateNestedItems={true}
        component={[Function]}
        expanderIcon={
          <FontIcon
            iconClassName="material-icons"
          >
            keyboard_arrow_down
          </FontIcon>
        }
        itemComponent="li"
        primaryText="My playlists"
        route="playlists"
      />,
      <Divider />,
      <ListItem
        activeClassName="md-text--theme-primary"
        animateNestedItems={true}
        component="div"
        expanderIcon={
          <FontIcon
            iconClassName="material-icons"
          >
            keyboard_arrow_down
          </FontIcon>
        }
        itemComponent="li"
        onClick={[Function]}
        primaryText="Log out"
      />,
    ]
  }
  persistentIcon={
    <FontIcon
      iconClassName="material-icons"
    >
      arrow_back
    </FontIcon>
  }
  position="left"
  tabletDrawerType="persistent"
  tabletMinWidth={768}
  temporaryIcon={
    <FontIcon
      iconClassName="material-icons"
    >
      menu
    </FontIcon>
  }
  toolbarThemeType="colored"
  toolbarTitle="Musicary"
  transitionEnterTimeout={300}
  transitionName="md-cross-fade"
/>
`;
